pkg/publish/default.go

```diff
@@ -217,6 +217,11 @@ func (d *defalt) Publish(ctx context.Context, br build.Result, s string) (name.R
 			return nil, err
 		}
 
+		err = remote.CheckPushPermission(tag, d.keychain, d.t)
```

Comment:
I think we need to do this at the publisher level for each (it might cause code duplication) because these are responsible for creating the tag and also we might not want to do this check for some of the publishers, such as layout publisher, etc.

Reply:
Thank you for comments.
I'm sorry to late reply.

I have seen [this PR](https://github.com/ko-build/ko/pull/964) and I would like to know if you have any suggestions for me. Thank you for correcting it in a good way.

I will close the #951  PR.